### DISCUSSION TITLE
Multilingual WiX WPF / C# custom Burn bootstrapper

### DISCUSSION THREAD

**Discussion by Peonshoes** [wixtoolset/issues#7876] [Questions]
Hi all,

I have created a custom Burn bootstrapper based on a C# Class Library using Visual Studio and it is working correctly. 
This is basically the example from Chapter 16 in the book 'WiX 3.6: A Developer's Guide to Windows Installer XML'.

I would now like to make the bootstrapper multi-lingual, where when run, the UI will automatically change to a supported language or if no supported language is found it will default to en-US. 
There is also a requirement for the user to be able to change the installation language via a ComboBox at the start of the installation.

I have a tried adding Resource Dictionaries and setting them in the XAML.cs code but without success.
Does anyone have any examples or information on how this can be achieved?

Notes:	
- The MSI that the bootstrapper calls is already localized.
- I am using Visual Studio 2019 , MVVM / Prism and WiX Toolset 3.11

Regards,

**Comment by nirbar** [comment]
It works the same as for any WPF project- set `Properties.Resources.Culture` to the culture you want before loading the window

**Reply by Peonshoes** [reply]
Thanks for the response nirbar,

I am currently setting the`ResourceDictionary` to change the language via StringResources, without success. 
Would setting `Properties.Resources.Culture` work with StringResources?
Should I be using `StringResources` or `resx` to store and manage the translations, what is the best practice for this?

When using StringResources files, I load the ResourceDictionary with the required StringResource, but I get a "`resource 'install' could not be resolved`" tooltip when using DynamicResource to retrieve the translation within the XAML file. 
e.g. : `<Button Command="{Binding InstallCommand}" Content="{DynamicResource install}"/>`

Any help and how to set the DynamicResource string would be appreciated. 
I can add code examples if required.

Thanks.
